src/huggingface_hub/commands/jobs.py

```diff
@@ -0,0 +1,471 @@
+
+import requests
+from dotenv import dotenv_values
+
+from huggingface_hub import HfApi
+from huggingface_hub.utils import logging
+
+from . import BaseHuggingfaceCLICommand
+
+
+logger = logging.get_logger(__name__)
+
+
+class JobsCommands(BaseHuggingfaceCLICommand):
+    @staticmethod
+    def register_subcommand(parser: _SubParsersAction):
+        jobs_parser = parser.add_parser("jobs", help="Commands to interact with your huggingface.co jobs.")
+        jobs_subparsers = jobs_parser.add_subparsers(help="huggingface.co jobs related commands")
+
+        # Register commands
+        InspectCommand.register_subcommand(jobs_subparsers)
+        LogsCommand.register_subcommand(jobs_subparsers)
+        PsCommand.register_subcommand(jobs_subparsers)
+        RunCommand.register_subcommand(jobs_subparsers)
+        CancelCommand.register_subcommand(jobs_subparsers)
+        UvCommand.register_subcommand(jobs_subparsers)
+
+
+class RunCommand(BaseHuggingfaceCLICommand):
+    @staticmethod
+    def register_subcommand(parser: _SubParsersAction) -> None:
+        run_parser = parser.add_parser("run", help="Run a Job")
+        run_parser.add_argument("image", type=str, help="The Docker image to use.")
+        run_parser.add_argument("-e", "--env", action="append", help="Set environment variables.")
+        run_parser.add_argument("-s", "--secret", action="append", help="Set secret environment variables.")
+        run_parser.add_argument("--env-file", type=str, help="Read in a file of environment variables.")
+        run_parser.add_argument("--secret-env-file", type=str, help="Read in a file of secret environment variables.")
+        run_parser.add_argument(
+            "--flavor",
+            type=str,
+            help="Flavor for the hardware, as in HF Spaces.",
+            default="cpu-basic",
+        )
+        run_parser.add_argument(
+            "--timeout",
+            type=str,
+            help="Max duration: int/float with s (seconds, default), m (minutes), h (hours) or d (days).",
+        )
+        run_parser.add_argument(
+            "-d",
+            "--detach",
+            action="store_true",
+            help="Run the Job in the background and print the Job ID.",
+        )
+        run_parser.add_argument(
+            "--token",
+            type=str,
+            help="A User Access Token generated from https://huggingface.co/settings/tokens",
+        )
+        run_parser.add_argument("command", nargs="...", help="The command to run.")
+        run_parser.set_defaults(func=RunCommand)
+
+    def __init__(self, args: Namespace) -> None:
+        self.image: str = args.image
+        self.command: List[str] = args.command
+        self.env: dict[str, Optional[str]] = {}
+        for env_value in args.env or []:
+            self.env.update(dotenv_values(stream=io.StringIO(env_value)))
+        if args.env_file:
+            self.env.update(dotenv_values(args.env_file))
```

Comment:
I feel that "env" should take precedence over "env_file" i.e. if someone hardcode a variable in its command line, it should not be overwritten by the file in case of conflict. Same for secrets.

(also possible to add a check to make `env` and `env_file` mutually exclusive)